internal/controller/eventing/integrationtests/controller/integration_test.go

```diff
@@ -153,32 +157,103 @@ func Test_CreateEventingCR(t *testing.T) {
+				// check if EPP deployment, HPA resources created and values are reflected including owner reference.
+				ensureEPPDeploymentAndHPAResources(t, tc.givenEventing, testEnvironment)
 				// TODO: ensure NATS Backend config is reflected. Done as subscription controller is implemented.
 			}
 
 			if tc.wantEnsureK8sObjects {
 				// check if EPP resources exists.
-				testEnvironment.EnsureEPPK8sResourcesExists(t, *tc.givenEventing)
+				ensureK8sResources(t, tc.givenEventing, testEnvironment)
+			}
+		})
+	}
+}
 
-				// check if the owner reference is set.
-				testEnvironment.EnsureEPPK8sResourcesHaveOwnerReference(t, *tc.givenEventing)
+func Test_CreateEventingCR_EventMesh(t *testing.T) {
+	t.Parallel()
 
-				// check if EPP resources are correctly created.
-				deployment, err := testEnvironment.GetDeploymentFromK8s(eventing.GetPublisherDeploymentName(*tc.givenEventing), givenNamespace)
-				require.NoError(t, err)
-				// K8s Services
-				testEnvironment.EnsureEPPPublishServiceCorrect(t, deployment, *tc.givenEventing)
-				testEnvironment.EnsureEPPMetricsServiceCorrect(t, deployment, *tc.givenEventing)
-				testEnvironment.EnsureEPPHealthServiceCorrect(t, deployment, *tc.givenEventing)
-				// ClusterRole
-				testEnvironment.EnsureEPPClusterRoleCorrect(t, *tc.givenEventing)
-				// ClusterRoleBinding
-				testEnvironment.EnsureEPPClusterRoleBindingCorrect(t, *tc.givenEventing)
+	testCases := []struct {
+		name                 string
+		givenEventing        *eventingv1alpha1.Eventing
+		givenDeploymentReady bool
+		wantMatches          gomegatypes.GomegaMatcher
+		wantEnsureK8sObjects bool
+	}{
+		{
+			name: "Eventing CR should have ready state when all deployment replicas are ready",
+			givenEventing: utils.NewEventingCR(
+				utils.WithEventingCRNamespace("test-namespace1"),
+				utils.WithEventMeshBackend("test-namespace1/test-secret-name"),
+				utils.WithEventingPublisherData(2, 2, "199m", "99Mi", "399m", "199Mi"),
+				utils.WithEventingEventTypePrefix("test-prefix"),
+			),
+			givenDeploymentReady: true,
+			wantMatches: gomega.And(
+				matchers.HaveStatusReady(),
+				//matchers.HaveNATSAvailableConditionAvailable(),
+				matchers.HavePublisherProxyReadyConditionDeployed(),
+				matchers.HaveFinalizer(),
+			),
+			wantEnsureK8sObjects: true,
+		},
+		{
+			name: "Eventing CR should have processing state when deployment is not ready yet",
+			givenEventing: utils.NewEventingCR(
+				utils.WithEventingCRNamespace("test-namespace2"),
+				utils.WithEventMeshBackend("test-namespace2/test-secret-name"),
+				utils.WithEventingPublisherData(2, 2, "199m", "99Mi", "399m", "199Mi"),
+				utils.WithEventingEventTypePrefix("test-prefix"),
+			),
+			wantMatches: gomega.And(
+				matchers.HaveStatusProcessing(),
+				//matchers.HaveNATSAvailableConditionAvailable(),
+				matchers.HavePublisherProxyReadyConditionProcessing(),
+				matchers.HaveFinalizer(),
+			),
+		},
```

Comment:
Should there be one more case for `Eventing CR should have error state ...`?

Reply:
I've really tried, but it is quite hard to get error case here as we don't mock k8s api server here.